memory/homenode.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM
+# Author: Geetika Moolchandani <geetika@linux.ibm.com>
+#
+
+
+import os
+import shutil
+from avocado import Test
+from avocado.utils import process, build, memory, genio, distro
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+class HomeNodeTest(Test):
+    """
+    Test Description
+    """
+
+    def copyutil(self, file_name):
+        shutil.copyfile(self.get_data(file_name),
+                        os.path.join(self.teststmpdir, file_name))
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.dist = distro.detect()
+
+        self.nr_pages = self.params.get('nr_pages', default=100)
+        self.map_type = self.params.get('map_type', default='private')
```

Comment:
Can you update the YAML file to also cover shared option as well? I see the testcase already has a code to cover shared option. Rest looks good.